lib/samson/secrets/vault_server.rb

```diff
@@ -36,7 +36,7 @@ class VaultServer < ActiveRecord::Base
         read_timeout: 2
       }.freeze
 
-      has_many :deploy_groups
+      has_many :deploy_groups, dependent: :nullify
```

Comment:
this is the default behavior too, so maybe that test already passes before changing the code ?

Reply:
Nope, this is not the default behavior, I Just tested, It's needed.